lib/matplotlib/axes/_axes.py

```diff
@@ -431,13 +503,28 @@ def inset_axes(self, bounds, *, transform=None, zorder=5,
         Parameters
         ----------
 
-        bounds : [x0, y0, width, height]
-            Lower-left corner of inset axes, and its width and height.
+        bounds : [x0, y0, width, height] or string.
+            If four-tupple: lower-left corner of inset axes, and its width and
+            height.
+
+            If a string, then locations such as "NE", "N", "NW", "W", etc,
+            of "upper right", "top", "upper left", etc (see `~.axes.legend`)
+            for codes.  (Note we do *not* support the numerical codes).
```

Comment:
As long as other functions  like `legend` support the number codes, I would support them everywhere. Once it is decided to deprectate them, I would deprecate them all, everywhere, in one single rush. One might silently allow them here, without mentionning them specifically though. <- my personal opinion (I'm not too inclined to open that discussion again as it will just never get to an end.)

Reply:
I hear you, but someone can add that as a followup PR if they really feel strongly about it.  Its a bunch of extra code for something we don't really want to support any longer.